### ISSUE TITLE
Missing api for rewriting input trees

### ISSUE THREAD

**Issue by fokot** [planet42/Laika#147]
I have input like
```
├── dir_1
│   ├── 01.md
│   ├── 02.md
│   └── 03.md
├── dir_2
│   ├── 01.md
│   ├── 02.md
│   └── 03.md
├── common
│   ├── 00-1.md
│   ├── 00-2.md
│   └── 00-3.md
└── default.template.html
```
And I want transformer to behave like if content of common was in `dir_1` and `dir_2` and skip generating `common` dir is it possible? So far I have found only one `fromDirectories` method which would merge content of few dirs together but I need more complex rewrite. Is there some api for this?

Also I have one more question here https://stackoverflow.com/questions/62353757/links-to-json-files
Thanks

**Comment by jenshalm** [comment]
That's a somewhat funny timing for this question. In the current release, which is just 2 weeks old, this is not possible yet (it kind of is, but only with pretty low-level APIs I won't have the time to walk you through).

In current master there is a new API since about a week that allows you to freely compose the inputs, including your scenario. If you are comfortable with building a snapshot yourself, I can show you an example.

**Comment by fokot** [comment]
Thank you @jenshalm for answer and cool library. I think easiest for me now would be to copy files to new temporary dir with right structure and generate it from there. Can you pls look at my question on SO too https://stackoverflow.com/questions/62353757/links-to-json-files?

**Comment by jenshalm** [comment]
The other issue turned out to be a bug that I just fixed, see #148 for details.
